Packages/com.unity.inputsystem/InputSystem/Editor/UITKAssetEditor/InputActionsEditorSettingsProvider.cs

```diff
@@ -27,20 +27,39 @@ public override void OnActivate(string searchContext, VisualElement rootElement)
             BuildUI();
         }
 
+        public override void OnDeactivate()
+        {
+            // Debug.Log("Project settings Editor destroyed, trying to save asset");
+            TrySaveAsset();
+        }
+
         private void OnStateChanged(InputActionsEditorState newState)
         {
             // Project wide input actions always auto save - don't check the asset auto save status
             InputActionsEditorWindowUtils.SaveAsset(m_State.serializedObject);
         }
 
+        private void TrySaveAsset()
+        {
+            if (m_State.serializedObject != null)
+            {
+                InputActionsEditorWindowUtils.SaveAsset(m_State.serializedObject);
+            }
+        }
+
         private void BuildUI()
         {
             m_StateContainer = new StateContainer(m_RootVisualElement, m_State);
-            m_StateContainer.StateChanged += OnStateChanged;
+            //TODO: remove this
+            // m_StateContainer.StateChanged += OnStateChanged;
             m_RootVisualElement.styleSheets.Add(InputActionsEditorWindowUtils.theme);
             new InputActionsEditorView(m_RootVisualElement, m_StateContainer, null);
             m_StateContainer.Initialize();
 
+            var mainElement = m_RootVisualElement.Q("actions-split-view");
+            mainElement.focusable = true;
+            mainElement.RegisterCallback<FocusOutEvent>(OnFocusLost);
```

Comment:
As I understand it this is only tracking the focus of the split view within the context of the Project Settings window, not OnFocus and OnFocusLost for the whole ProjectSettings window which is what I would expect we are interested in? Seems like we need additional work here to track the top-level window?

Ignore comment above, it seems as as long the actual split-view or its sub-content get the focus it will also trigger on Project Settings loosing focus to an external (non-Unity) window. Seems to be doing what is expected as-is since it would be impossible to edit anything unless the split-view (or child) gets focus anyway.